### PR TITLE
feat(wallet): Implement KYC - personal details-ID proof

### LINKED ISSUE #50: Create KYC Wallet - Validate User Account (FE)
Create the KYC Wallet forms, to validate user account:
image

### PR DESCRIPTION
**Pull request by Tymmmy**
### Context
- fixes #50

### Changes
Created form for Personal Information KYC, and part of ID proof KYC.
Rapyd API connection and communication with backend still needed to see finer details here.
Bank details not yet implemented

### CODE REVIEW

File: packages/wallet/src/components/kyc/IdentificationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+              htmlFor={idType.id}
+              className="min-w-[50px] cursor-pointer rounded-xl border border-brand-turqoise bg-white p-2 text-center text-base font-light text-brand-turqoise shadow-md peer-checked:border-brand-orange peer-checked:text-brand-orange sm:min-w-[100px]"
+            >
+              {idType.label}
+            </label>
+          </div>
+        ))}
+      </div>
+      <div className="flex">
+        <FileUpload label="Front side" />
```
Comment (Tymmmy) [inline packages/wallet/src/components/kyc/IdentificationForm.tsx:60]: The File Uploads/Labels might also come from backend depending on what id type is selected

File: packages/wallet/src/components/layouts/AuthLayout.tsx [hunk 2]
```diff
@@ -1,35 +1,52 @@
...
               src={imageSrc}
               alt={image}
               quality={100}
             />
           </div>
-          <div className="min-h-full flex-1 bg-brand-blue-1/70 py-10">
+          <div
+            className={cx(
+              'min-h-full flex-1 py-10',
+              background === 'brand-green'
```
Comment (raducristianpopa) [inline packages/wallet/src/components/layouts/AuthLayout.tsx:45]:
```suggestion
              background === 'green'
```

File: packages/wallet/src/components/layouts/AuthLayout.tsx [hunk 3]
```diff
@@ -1,35 +1,52 @@
...
   const imageSrc = `/${IMAGES[image]}`
   return (
     <>
       <div className="h-full">
         <div className="flex min-h-full">
-          <div className="relative hidden w-0 flex-1 bg-brand-blue-1/70 md:block">
+          <div
+            className={cx(
+              'relative hidden w-0 flex-1 md:block',
+              background === 'brand-green'
```
Comment (raducristianpopa) [inline packages/wallet/src/components/layouts/AuthLayout.tsx:29]:
```suggestion
              background === 'green'
```

File: packages/wallet/src/components/HeaderLogo.tsx [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
+      />
+      <h1
+        className={cx(
+          'hidden space-x-4 text-5xl font-semibold md:mt-10 md:block',
+          type === 'kyc' ? 'text-brand-green-3' : 'text-brand-turqoise'
+        )}
+      >
         {header}
       </h1>
     </>
```
Comment (raducristianpopa) [inline packages/wallet/src/components/HeaderLogo.tsx:36]:
```suggestion
  <div className="flex flex-col-reverse items-center md:flex-col">
      <Logo className={cx('h-28 w-28 flex-shrink-0')} />
      <h1
        className={cx(
          'mb-10 block space-x-4 text-xl md:mb-0 md:mt-10 md:text-5xl md:font-semibold',
          type === 'kyc' ? 'text-brand-green-3' : 'text-brand-turqoise'
        )}
      >
        {header}
      </h1>
    </div>
```

We can do something like this and get rid of one of the header elements. ([`flex-col-reverse`](https://tailwindcss.com/docs/flex-direction#column-reversed))

File: packages/wallet/src/components/kyc/PersonalDetailsForm.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          label="Last name"
+        />
+      </div>
+      <Select
+        name="country"
+        setValue={personalDetailsForm.setValue}
+        defaultValue={countries[0]}
+        error={personalDetailsForm.formState.errors.country?.message}
+        options={countries}
+        label="Asset"
```
Comment (raducristianpopa) [inline packages/wallet/src/components/kyc/PersonalDetailsForm.tsx:53]:
```suggestion
        label="Country"
```

File: packages/wallet/src/components/kyc/context.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  setTab: (tab: number) => void
+}
+
+const KYCFormContext = createContext<KYCFormContextProps | null>(null)
+
+export const useKYCFormContext = () => {
+  const formContext = useContext(KYCFormContext)
+
+  if (!formContext) {
+    throw new Error('"useFormContext" is used outside the FormContextProvider.')
```
Comment (raducristianpopa) [inline packages/wallet/src/components/kyc/context.tsx:14]:
```suggestion
    throw new Error('"useKYCFormContext" is used outside the KYCFormContextProvider.')
```

File: packages/wallet/src/components/kyc/index.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import { TabsNavigation, Tabs } from './Tabs'
+
+export const KYC = () => {
+  return (
+    <>
+      <KYCFormContextProvider>
+        <Tabs />
+        <TabsNavigation />
+      </KYCFormContextProvider>
+    </>
```
Comment (raducristianpopa) [inline packages/wallet/src/components/kyc/index.tsx:11]:
```suggestion
      <KYCFormContextProvider>
        <Tabs />
        <TabsNavigation />
      </KYCFormContextProvider>
```

File: packages/wallet/src/components/HeaderLogo.tsx (hunk 4 above)
Comment (Tymmmy) [reply packages/wallet/src/components/HeaderLogo.tsx:36]: Nice solution

File: packages/wallet/src/components/kyc/Tabs.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        <Button aria-label="submit" type="submit" className="mx-auto">
+          Submit
+        </Button>
+      )}
+      {tab < tabs.length - 1 && (
+        <button aria-label="login" onClick={() => setTab(tab + 1)}>
+          <Play />
+        </button>
+      )}
+    </div>
```
[the remark this answers is not in the document]
Comment (Tymmmy) [reply packages/wallet/src/components/kyc/Tabs.tsx:127]: A row-reverse did the trick :)

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**